dali/pipeline/operators/arithmetic/arithmetic_test.cc

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/operators/arithmetic/arithmetic.h"
+#include "dali/pipeline/operators/arithmetic/arithmetic_meta.h"
+#include "dali/pipeline/pipeline.h"
+#include "dali/test/dali_operator_test.h"
+
+namespace dali {
+
+TEST(ArithmOps, ExpressionTree) {
+  std::string expr = "mul(&0 &1)";
+  auto result = ParseExpressionString(expr);
+  auto &result_ref = *result;
+  ASSERT_FALSE(result_ref.IsLeaf());
+  ASSERT_EQ(result_ref.GetSubexpressionCount(), 2);
+  ASSERT_EQ(result_ref.GetOp(), "mul");
+  ASSERT_TRUE(result_ref[0].IsLeaf());
+  ASSERT_TRUE(result_ref[1].IsLeaf());
+  ASSERT_EQ(result_ref[0].GetMappedInput(), 0);
+  ASSERT_EQ(result_ref[1].GetMappedInput(), 1);
+}
+
+TEST(ArithmOps, ExpressionTreeComplex) {
+  std::string expr = "div(sub(&42 &2) &1)";
+  auto result = ParseExpressionString(expr);
+  auto &result_ref = *result;
+  ASSERT_FALSE(result_ref.IsLeaf());
+  ASSERT_EQ(result_ref.GetSubexpressionCount(), 2);
+  EXPECT_EQ(result_ref.GetOp(), "div");
+  ASSERT_FALSE(result_ref[0].IsLeaf());
+  ASSERT_EQ(result_ref[0].GetSubexpressionCount(), 2);
+  EXPECT_EQ(result_ref[0].GetOp(), "sub");
+  EXPECT_TRUE(result_ref[0][0].IsLeaf());
+  EXPECT_EQ(result_ref[0][0].GetMappedInput(), 42);
+  EXPECT_TRUE(result_ref[0][1].IsLeaf());
+  EXPECT_EQ(result_ref[0][1].GetMappedInput(), 2);
+  EXPECT_TRUE(result_ref[1].IsLeaf());
+  EXPECT_EQ(result_ref[1].GetMappedInput(), 1);
+}
+
+// namespace {
+//   bool operator==(const SampleExtent &l, const SampleExtent &r) {
+//     return l.extent_idx == r.extent_idx && l.extent_size == l.extent_size &&
+//            l.sample_idx == r.sample_idx && l.task_idx == r.task_idx;
+//   }
+// }  // namespace
+
+// TEST(ArithmOps, GetSampleCover) {
```

Comment:
will you enable or remove?

Reply:
Enable when I convince gtest to compare them.